src/components/TextLink.tsx

```diff
@@ -0,0 +1,83 @@
+import React, {FC, ForwardedRef, KeyboardEventHandler, MouseEventHandler, ReactElement} from 'react';
+import {GestureResponderEvent, Text as RNText, TextStyle} from 'react-native';
+import styles from '@styles/styles';
+import * as Link from '@userActions/Link';
+import CONST from '@src/CONST';
+import Text from './Text';
+
+type TextLinkProps = {
+    /** Link to open in new tab */
+    href: string;
+
+    /** Text content child */
+    children: ReactElement;
+
+    /** Additional style props */
+    style?: TextStyle;
+
+    /** Overwrites the default link behavior with a custom callback */
+    onPress?: () => void;
+
+    /** Callback that is called when mousedown is triggered */
+    onMouseDown?: MouseEventHandler;
+
+    /** A ref to forward to text */
+    forwardedRef: ForwardedRef<RNText>;
+};
+
+function TextLink({href, children, style, onPress, onMouseDown = (event) => event.preventDefault(), forwardedRef, ...props}: TextLinkProps) {
```

Comment:
According to typescript guidelines forwardRef should be handled a bit differently. Please remove `forwardedRef: ForwardedRef<RNText>;` prop and move it to the second parameter like this:
 ```suggestion
function TextLink({href, children, style, onPress, onMouseDown = (event) => event.preventDefault(), ...props}: TextLinkProps, ref: ForwardedRef<RNText>) {
```